### ISSUE TITLE
questions about unpooling layer

### ISSUE THREAD

**Issue by jinluyang** [weiliu89/caffe#914]
Please use the [caffe-users list](https://groups.google.com/forum/#!forum/caffe-users) for usage, installation, or modeling questions, or other requests for help.
_Do not post such requests to Issues._ Doing so interferes with the development of Caffe.

Please read the [guidelines for contributing](https://github.com/BVLC/caffe/blob/master/CONTRIBUTING.md) before submitting this issue.

### Issue summary
In the origin paper I found the unpooling layer used to get feature maps from a global vector. So its input should have h=1 and w=1. But I am confused to find that in the code the unpooling layer support more than that. Can it be used in other occasions? Besides, I find the DIV and REP method of unpool similar to the backprop process of average pooling, just that they have to divide a mask after that, am I right? But the paper didn't mention about the mask. I hope to get a better understanding of this layer, thank you!

### Steps to reproduce

If you are having difficulty building Caffe or training a model, please ask the caffe-users mailing list. If you are reporting a build error that seems to be due to a bug in Caffe, please attach your build configuration (either Makefile.config or CMakeCache.txt) and the output of the make (or cmake) command.

### Your system configuration
Operating system:
Compiler:
CUDA version (if applicable):
CUDNN version (if applicable):
BLAS:
Python or MATLAB version (for pycaffe and matcaffe respectively):
